pkg/capabilities/capabilities.go

```diff
@@ -105,10 +105,33 @@ type CallbackExecutable interface {
 	Execute(ctx context.Context, callback chan<- CapabilityResponse, request CapabilityRequest) error
 }
 
+type Validatable interface {
+	// GetConfigJsonSchema will create a values-wrapped JSON schema for the "config" field of the capability.
+	//
+	// This allows for more comprehensive validation of a workflow yaml file since
+	// after dynamically resolving the capability, the workflow engine can validate
+	// the config field against the schema.
+	//
+	// NOTE: What to do with method calls like:
+	//
+	//  - `RegisterToWorkflow` and `UnregisterFromWorkflow`
+	//  - `RegisterTrigger` and `UnregisterTrigger`
+	//
+	// They take in a request object that has a `Config` field that may conflate
+	// with the `Config` field within `Execute`.
+	GetRequestConfigJSONSchema() *CapabilityResponse
```

Comment:
@HenryNguyen5 No objection to using JSONSchema for validation, I think that's great, but I'm a bit concerned about the API here.

Since this is truly generic (i.e. all capabilities) will provide us with a JSONSchema to validate against, I think we can be very concrete with our API and not use something like a CapabilityResponse. Some concrete JSONSchema type would be fine 👍

Reply:
On your comment about multiple method calls, I think having a single schema per call is fine 👍  or alternatively having a single method which returns a map of method names to schemas.

Reply:
Ah, the name feels pretty clunky too!

Getters in golang usually omit the GetX prefix, so would rename this to InputsSchema(), and maybe add a ConfigSchema() for config fields.